dashboard/src/main/java/com/google/cloud/tools/opensource/dashboard/DashboardMain.java

```diff
@@ -157,28 +165,38 @@ static Path generateVersionIndex(String groupId, String artifactId, List<String>
   }
 
   @VisibleForTesting
-  static Path generate(String bomCoordinates)
+  static Path generate(
+      String bomCoordinates, DependencyMediationAlgorithm dependencyMediationAlgorithm)
       throws IOException, TemplateException, RepositoryException, URISyntaxException {
-    Path output = generate(Bom.readBom(bomCoordinates));
+    Path output = generate(Bom.readBom(bomCoordinates), dependencyMediationAlgorithm);
     System.out.println("Wrote dashboard for " + bomCoordinates + " to " + output);
     return output;
   }
 
   @VisibleForTesting
-  static Path generate(Path bomFile)
-      throws IOException, TemplateException, URISyntaxException, MavenRepositoryException {
+  static Path generate(Path bomFile, DependencyMediationAlgorithm dependencyMediationAlgorithm)
+      throws IOException, TemplateException, URISyntaxException, MavenRepositoryException,
+          InvalidVersionSpecificationException {
     checkArgument(Files.isRegularFile(bomFile), "The input BOM %s is not a regular file", bomFile);
     checkArgument(Files.isReadable(bomFile), "The input BOM %s is not readable", bomFile);
-    Path output = generate(Bom.readBom(bomFile));
+    Path output = generate(Bom.readBom(bomFile), dependencyMediationAlgorithm);
     System.out.println("Wrote dashboard for " + bomFile + " to " + output);
     return output;
   }
 
-  private static Path generate(Bom bom) throws IOException, TemplateException, URISyntaxException {
+  private static Path generate(Bom bom, DependencyMediationAlgorithm dependencyMediationAlgorithm)
+      throws IOException, TemplateException, URISyntaxException,
+          InvalidVersionSpecificationException {
 
     ImmutableList<Artifact> managedDependencies = bom.getManagedDependencies();
 
-    ClassPathResult classPathResult = classPathBuilder.resolve(managedDependencies, false);
+    DependencyMediation dependencyMediation =
+        dependencyMediationAlgorithm == DependencyMediationAlgorithm.MAVEN
+            ? DependencyMediation.MAVEN
+            : GradleDependencyMediation.withEnforcedPlatform(bom);
+
+    ClassPathResult classPathResult =
+        classPathBuilder.resolve(managedDependencies, false, dependencyMediation);
```

Comment:
This is the goal of this PR. Generating a class path through Gradle's dependency mediation algorithm.